integration/utils_test.go

```diff
@@ -350,3 +351,7 @@ func parseRun(args []string) (*runconfig.Config, *runconfig.HostConfig, *flag.Fl
 	cmd.Usage = nil
 	return runconfig.Parse(cmd, args)
 }
+
+func getDaemon(eng *engine.Engine) *daemon.Daemon {
+	return eng.HackGetGlobalVar("httpapi.daemon").(*daemon.Daemon)
```

Comment:
just a quick comment, I didn't read the PR thoroughly, but we introduced `HackGetGlobalVar` when we introduced jobs. I'm surprised to see this as the goal of this PR is to remove jobs.